service/cluster/cluster.go

```diff
@@ -290,7 +298,22 @@ func (s *Service) getPeers() ([]string, error) {
 	if !s.opts.EnableClustering || s.opts.DiscoverPeers == nil {
 		return nil, nil
 	}
-	return s.opts.DiscoverPeers()
+
+	peers, err := s.opts.DiscoverPeers()
+	if err != nil {
+		return nil, err
+	}
+
+	// Here we return the entire list because we can't take a subset.
+	if s.opts.ClusterMaxInitJoinPeers == 0 || len(peers) < s.opts.ClusterMaxInitJoinPeers {
+		return peers, nil
+	}
+
+	// We shuffle the list and return only a subset of the peers.
+	s.randGen.Shuffle(len(peers), func(i, j int) {
+		peers[i], peers[j] = peers[j], peers[i]
+	})
+	return peers[:s.opts.ClusterMaxInitJoinPeers], nil
```

Comment:
So this change would not only affect the _initial_ set of peers joined, but also the peers we try to join during a refresh (which is what we want).

Could we update the parameter name, docs and PR title to refer to that?

Reply:
thanks for pointing it out, at first I made the change only for the startup but then changed the implementation and overlooked the refresh. Good that we also want it in the refresh then, I will update everything to refer to that